synapse/lib/stormlib/smtp.py

```diff
@@ -0,0 +1,183 @@
+            'attach': self.send,
+            'headers': self.headers,
+            'recipients': self.recipients,
+        })
+
+        self.gtors['text'] = self._getEmailText
+        self.gtors['html'] = self._getEmailHtml
+        self.stors['text'] = self._setEmailText
+        self.stors['html'] = self._setEmailHtml
+
+        self.stors['sender'] = self._setSenderEmail
+        self.gtors['sender'] = self._getSenderEmail
+
+    async def _setSenderEmail(self, valu):
+        # TODO handle inet:email and ps:contact Nodes
+        self.sender = await s_stormtypes.tostr(valu)
+
+    async def _getSenderEmail(self):
+        return self.sender
+
+    async def _setEmailText(self, text):
+        self.bodytext = await s_stormtypes.tostr(text)
+
+    async def _setEmailHtml(self, html):
+        self.bodyhtml = await s_stormtypes.tostr(html)
+
+    async def _getEmailText(self):
+        return self.bodytext
+
+    async def _getEmailHtml(self):
+        return self.bodyhtml
+
+    #TODO
+    #async def attach(self, sha256, name, mime):
+
+    async def send(self, host, port=25, user=None, passwd=None, usetls=False, starttls=False, timeout=60):
+
+        self.runt.confirm(('storm', 'inet', 'smtp', 'send'))
+
+        try:
+            if self.bodytext is None and self.bodyhtml is None:
+                mesg = 'The storm:smtp:message has no HTML or text body.'
+                raise s_exc.StormRuntimeError(mesg=mesg)
+
+            host = await s_stormtypes.tostr(host)
+            port = await s_stormtypes.toint(port)
+            usetls = await s_stormtypes.tobool(usetls)
+            starttls = await s_stormtypes.tobool(starttls)
+
+            timeout = await s_stormtypes.toint(timeout)
+
+            user = await s_stormtypes.tostr(user, noneok=True)
+            passwd = await s_stormtypes.tostr(passwd, noneok=True)
+
+            message = MIMEMultipart('alternative')
+
+            if self.bodytext is not None:
+                message.attach(MIMEText(self.bodytext, 'plain', 'utf-8'))
+
+            if self.bodyhtml is not None:
+                message.attach(MIMEText(self.bodyhtml, 'html', 'utf-8'))
+
+            for name, valu in self.headers.items():
+                message[await s_stormtypes.tostr(name)] = await s_stormtypes.tostr(valu)
+
+            recipients = [await s_stormtypes.tostr(e) for e in self.recipients]
+
+            futu = aiosmtplib.send(message,
+                    port=port,
+                    hostname=host,
+                    sender=self.sender,
+                    recipients=recipients,
+                    use_tls=usetls,
+                    start_tls=starttls,
+                    username=user,
+                    password=passwd)
+
+            await asyncio.wait_for(futu, timeout=timeout)
+
+        except asyncio.CancelledError: # pragma: no cover
+            raise
+
+        except Exception as e:
+            return (False, s_common.excinfo(e))
+
+        return (True, None)
```

Comment:
aiosmtplib.send does return a dict, might be useful to return this to the user

Reply:
Looks like it includes heavy objects that might need translation, but i'll change the return convention for our send() API to returning `(True, {})` on success so we can be flexible in the future.